java/core/src/main/java/co/worklytics/psoxy/gateway/ConfigService.java

```diff
@@ -38,7 +37,29 @@ default Boolean noCache() {
          * @return whether this property is limited to being set via environment variables only
          */
         default boolean isEnvVarOnly() {
-            return false;
+            return getSupportedSource() == SupportedSource.ENV_VAR;
+        }
+
+        default SupportedSource getSupportedSource() {
+            return SupportedSource.ENV_VAR_OR_REMOTE;
+        }
+
+        enum SupportedSource {
+
+            /**
+             * config value MUST be set via environment variable; remote source is NOT supported
+             */
+            ENV_VAR,
+
+            /**
+             * config value MUST be set in the remote source; env var is NOT Supported
+             */
+            REMOTE,
+
+            /**
+             * config value may be set either via environment variable or remote config, with former taking precedence
+             */
+            ENV_VAR_OR_REMOTE,
```

Comment:
nitpick: I'd probably remove this and use a List of supported values, with order implicit in list.
define a constant here
`List<SupportedSource> ENV_VAR_OR_REMOTE = List.of(ENV_VAR, REMOTE)`

Reply:
said this, I don't think this will grow... so fine as is

Reply:
yeah. I don't think it will grow.

I do think having these as some kind of `Option`, with a variadic constructor `(Option...)` would be more readable - the default case (either) not so straightforward though.

Going to leave as-is for now; easy future change.